From 9abb15d070d9ac84e5aea41557e196e6796f5f10 Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Wed, 3 Sep 2014 21:18:21 -0400
Subject: [platform] toshiba: re-enable acpi hotkeys after suspend to disk

Message-id: <1409779101-25646-1-git-send-email-benjamin.tissoires@redhat.com>
Patchwork-id: 91212
O-Subject: [RHEL7.1 PATCH] platform/x86: toshiba: re-enable acpi hotkeys after suspend to disk
Bugzilla: 1107814
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1107814
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7915156
Upstream: in the platform-driver maintainer tree, should show up in
          3.17 or 3.18
Tested by me on a Toshiba Tecra Z40

commit e7fdb762b9e1e10c3271e47723b2003330829ddf in
git://git.infradead.org/users/dvhart/linux-platform-drivers-x86.git

Author: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date:   Tue Sep 2 14:04:19 2014 -0400

    platform/x86: toshiba: re-enable acpi hotkeys after suspend to disk

    On the Toshiba Tecra Z40, after a suspend-to-disk, some FN hotkeys
    driven by toshiba_acpi are not functional.

    Calling the ACPI object ENAB on resume makes them back alive.

    Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
    Acked-by: Matthew Garrett <matthew.garrett@nebula.com>
    Signed-off-by: Darren Hart <dvhart@linux.intel.com>

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/platform/x86/toshiba_acpi.c b/drivers/platform/x86/toshiba_acpi.c
index eb3467e..94b323f 100644
--- a/drivers/platform/x86/toshiba_acpi.c
+++ b/drivers/platform/x86/toshiba_acpi.c
@@ -1311,9 +1311,16 @@ static int toshiba_acpi_resume(struct device *device)
 {
  struct toshiba_acpi_dev *dev = acpi_driver_data(to_acpi_device(device));
  u32 result;
+ acpi_status status;
+
+ if (dev->hotkey_dev) {
+  status = acpi_evaluate_object(dev->acpi_dev->handle, "ENAB",
+    NULL, NULL);
+  if (ACPI_FAILURE(status))
+   pr_info("Unable to re-enable hotkeys\n");
 
- if (dev->hotkey_dev)
   hci_write1(dev, HCI_HOTKEY_EVENT, HCI_HOTKEY_ENABLE, &result);
+ }
 
  return 0;
 }
-- 
1.7.1